kv-storage/src/rocksdb_kv/mod.rs

```diff
@@ -0,0 +1,300 @@
+        let checkpoint_db_dir = Temp::new_dir().unwrap();
+        {
+            let db = DB::open_default(origin_path.to_str().unwrap()).unwrap();
+            let checkpoint_db = checkpoint::Checkpoint::new(&db).unwrap();
+
+            checkpoint_db
+                .create_checkpoint(current_db_dir.to_path_buf().as_path().join("db"))
+                .unwrap();
+            checkpoint_db
+                .create_checkpoint(checkpoint_db_dir.to_path_buf().as_path().join("db"))
+                .unwrap();
+        }
+
+        Ok(RocksDB {
+            db: DB::open_default(current_db_dir.to_path_buf().as_path().join("db")).unwrap(),
+            origin_path,
+            current_db_dir,
+            checkpoint_db_dir,
+        })
+    }
+
+    async fn open(path: &str) -> Result<Self, super::Error>
+    where
+        Self: Sized,
+    {
+        let origin_path = PathBuf::from(path);
+        let current_db_dir = Temp::new_dir().unwrap();
+        let checkpoint_db_dir = Temp::new_dir().unwrap();
+        {
+            let db = DB::open_default(origin_path.to_str().unwrap()).unwrap();
+            let checkpoint_db = checkpoint::Checkpoint::new(&db).unwrap();
+
+            checkpoint_db
+                .create_checkpoint(current_db_dir.to_path_buf().as_path().join("db"))
+                .unwrap();
+            checkpoint_db
+                .create_checkpoint(checkpoint_db_dir.to_path_buf().as_path().join("db"))
+                .unwrap();
+        }
+
+        Ok(RocksDB {
+            db: DB::open_default(current_db_dir.to_path_buf().as_path().join("db")).unwrap(),
+            origin_path,
+            current_db_dir,
+            checkpoint_db_dir,
+        })
+    }
+
+    async fn commit_checkpoint(&mut self) -> Result<(), super::Error> {
+        let new_checkpoint_db_dir = Temp::new_dir().unwrap();
+        let checkpoint_db = checkpoint::Checkpoint::new(&self.db).unwrap();
+
+        DB::destroy(&Options::default(), self.origin_path.to_str().unwrap()).unwrap();
+        checkpoint_db
+            .create_checkpoint(self.origin_path.to_str().unwrap())
+            .unwrap();
+        checkpoint_db
+            .create_checkpoint(new_checkpoint_db_dir.to_path_buf().as_path().join("db"))
+            .unwrap();
+        self.checkpoint_db_dir = new_checkpoint_db_dir;
+
+        Ok(())
+    }
+
+    async fn revert_to_latest_checkpoint(&mut self) -> Result<(), super::Error> {
+        let new_current_db_dir = Temp::new_dir().unwrap();
+        let new_checkpoint_db_dir = Temp::new_dir().unwrap();
+        {
+            let new_db =
+                DB::open_default(self.checkpoint_db_dir.to_path_buf().as_path().join("db"))
+                    .unwrap();
+            let checkpoint_db = checkpoint::Checkpoint::new(&new_db).unwrap();
+
+            checkpoint_db
+                .create_checkpoint(new_current_db_dir.to_path_buf().as_path().join("db"))
+                .unwrap();
```

Comment:
Don't unwrap, but handle it and propagate using `?`

Reply:
fix